### PR TITLE
Issue 6945: Lombok breaks compilation with JDK 11.0.17

### LINKED ISSUE #6945: Lombok breaks compilation with JDK 11.0.17
**Describe the bug**
After the OS upgraded the JDK to 11.0.17 version, me (and others) started to observe this Lombok parsing issue that breaks compilation:
```
Exception while resolving: NODE LOCAL (class com.sun.tools.javac.tree.JCTree$JCVariableDecl) @val()
final ___Lombok_VAL_Attrib__ latestInfo = (StreamSegmentInformation)store.getStreamSegmentInfo(segmentName, TIMEOUT).join()(SimpleFileObject[/home/raul/Documents/workspace/PRAVEGA-2142/pravega/segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java])
java.lang.NullPointerException
        at jdk.compiler/com.sun.tools.javac.comp.Flow$FlowAnalyzer.visitApply(Flow.java:1235)
```

See https://github.com/projectlombok/lombok/issues/3292 

**To Reproduce**
See https://github.com/projectlombok/lombok/issues/3292

**Screenshots**
See https://github.com/projectlombok/lombok/issues/3292

**Additional information**
The problem seems to be related to some specific usage of Lombok with the lates JDK 11 version. Neither GA nor Jenkins builds are failing, as they are still using 11.0.16 version of JDK, but the problem is likely to happen as soon as JDK is upgraded.

### PR DESCRIPTION
**Pull request by shshashwat**
Signed-off-by: Shashwat Sharma <shashwat_sharma@dell.com>

**Change log description**  
After the OS upgraded the JDK to 11.0.17 version, the build stated failing with an observation with Lombok parsing issue that breaks compilation:
```
Exception while resolving: NODE LOCAL (class com.sun.tools.javac.tree.JCTree$JCVariableDecl) @val()
final ___Lombok_VAL_Attrib__ latestInfo = (StreamSegmentInformation)store.getStreamSegmentInfo(segmentName, TIMEOUT).join()(SimpleFileObject[/home/raul/Documents/workspace/PRAVEGA-2142/pravega/segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java])
java.lang.NullPointerException
        at jdk.compiler/com.sun.tools.javac.comp.Flow$FlowAnalyzer.visitApply(Flow.java:1235)
```
Another compilation issue was observed with JDK upgrade was 
```
compiler message file broken: key=compiler.misc.msg.bug arguments=11.0.17, {1}, {2}, {3}, {4}, {5}, {6}, {7}
java.lang.NullPointerException

```

which repeatedly kept the compiler to fail which eventually turned out to be an issue in OpenJDK version. 
**Purpose of the change**  
Fixes #6945 

**What the code does**  
Replaces all lombok  ``` val ``` with actual datatype after identifying all affected area.
Handle the JDK issue by refactoring the impacted code blocks after identification

**How to verify it**  
There shouldn't be any compilation issue - 
Prior to changes compiler message for the command ``` ./gradlew clean :segmentstore:storage:impl:compileTestJava ``` was:-
``` compiler message file broken: key=compiler.misc.msg.bug arguments=11.0.17, {1}, {2}, {3}, {4}, {5}, {6}, {7} ```
After the changes:- 
```
 Build JVM Version is : 11.0.17 (Ubuntu 11.0.17+8-post-Ubuntu-1ubuntu220.04)
BUILD SUCCESSFUL

 ```
The build should pass
All tests should pass
System test should pass

### CODE REVIEW

File: client/src/main/java/io/pravega/client/segment/impl/ConditionalOutputStreamImpl.java [hunk 1]
```diff
@@ -103,9 +102,9 @@ public boolean write(ByteBuffer data, long expectedOffset) throws SegmentSealedE
                             }
                         }
                         long requestId = client.getFlow().getNextSequenceNumber();
-                        val request = new ConditionalAppend(writerId, appendSequence, expectedOffset,
+                        ConditionalAppend request = new ConditionalAppend(writerId, appendSequence, expectedOffset,
```
Comment (sachin-j-joshi) [inline client/src/main/java/io/pravega/client/segment/impl/ConditionalOutputStreamImpl.java:105]: please add final

File: segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/AsyncStorageWrapperTests.java [hunk 2]
```diff
@@ -191,24 +193,29 @@ public void testConcurrencySameSegment() throws Exception {
...
         invoked.put(op1, new ReusableLatch());
         invoked.put(op2, new ReusableLatch());
         waitOn.put(op1, new ReusableLatch());
         waitOn.put(op2, new ReusableLatch());
 
-        val innerStorage = new TestStorage((operation, segment) -> {
+        final TestStorage innerStorage = new TestStorage((operation, segment) -> {
             invoked.get(operation).release();
-            Exceptions.handleInterrupted(() -> waitOn.get(operation).await());
+            try {
```
Comment (eolivelli) [inline segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/AsyncStorageWrapperTests.java:205]: why do we need to change this ?

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: client/src/main/java/io/pravega/client/segment/impl/ConditionalOutputStreamImpl.java (hunk 1 above)
Comment (shshashwat) [reply client/src/main/java/io/pravega/client/segment/impl/ConditionalOutputStreamImpl.java]: added at all places

**Review by RaulGracia — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
+1

**Comment by eolivelli** [comment]
thank you @RaulGracia for your clarification

**Comment by shshashwat** [comment]
System test TPC1-4850 succeeded

**Pull request merged**